client.go

```diff
@@ -102,7 +104,7 @@ func NewClient(config *Config) (client *Client, err error) {
 
 	tr := http.DefaultTransport.(*http.Transport)
 	tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: config.SkipSslValidation}
-	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: tr})
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, &http.Client{Transport: tr, Timeout: config.Timeout})
```

Comment:
ok, the client *might get replaced* if the `TokenSource` is `nil`. While this is not the case here, I would consider it dangerous to rely on that.

To be sure that the configured `Timeout` value is honored, it is written to whatever has become the `config.HttpClient` in [line 130](https://github.com/cloudfoundry-community/go-cfclient/pull/81/files#diff-d0b7f86cea13f90d3cc0cc2338da8d5fR130).
Or am I still missing something?